Core/clim-basic/geometry/region-utilities.lisp

```diff
@@ -0,0 +1,1204 @@
+(in-package #:climi)
+
+
+;;; point and angle utilities
+
+;;; CLIM "native" coordinate system is left-handed (y grows down).
+;;; Angles are specified to grow in the counter-clockwise direction
+;;; (disregarding of the coordinate system), so we need to invert the
+;;; y coordinate in a call to atan.
+
+(defun atan* (x y)
+  ;; atan like we need it.
+  ;; For us, phi=0 along the x axis and angles are always between 0 and 2pi.
+  (let ((r (atan (float y pi) x)))
+    (if (< r 0) (+ r (* 2 pi)) r)))
+
+(declaim (inline angle-contains-angle-p))
+(defun angle-contains-angle-p (alpha omega delta)
+  (if (< alpha omega)
+      (coordinate-between* alpha delta omega)
+      (or (coordinate<= alpha delta)
+          (coordinate<= delta omega))))
```

Comment:
I found the old name and lambda list ` %angle-between-p (angle start-angle end-angle)` much clearer.

Reply:
this rename is more consistent with angle-contains-point-p so I'd rather keep it this way (I find it better despite coming up first with %angle-between-p)

Reply:
Sorry, but I think it is really unclear. Firstly because it is not clear how an angle can "contain" another angle, secondly because `alpha omega delta` do not indicate which role the respective parameter plays.

I can understand that you would like something using the `X-contains-Y-p` shape. How about circle-segment contains angle or arc contains angle with parameters `angle` and e.g. `arc-start-angle` and `arc-end-angle`?

Reply:
pushed as a separate commit (last one)